drivers/led/ltc3220/ltc3220.c

```diff
@@ -0,0 +1,537 @@
+ * @brief Resets the device using the RST pin
+ * @param device - the device structure.
+ * @return 0 in case of success, negative error code otherwise.
+*******************************************************************************/
+int ltc3220_reset(struct ltc3220_dev *device)
+{
+	int ret;
+	ret = no_os_gpio_direction_output(device->gpio_rst_desc, NO_OS_GPIO_LOW);
+	if (ret)
+		return ret;
+
+	no_os_udelay(LTC3220_RESET_DELAY_USEC); /* at least 20ns */
+
+	ret = no_os_gpio_set_value(device->gpio_rst_desc, NO_OS_GPIO_HIGH);
+	if (ret)
+		return ret;
+
+	device->blink_config = blink_reset;
+	device->command_config = command_reset;
+	device->gradation_config = grad_reset;
+	for (int i = 0; i < LTC3220_REG_END_ULED; i++) {
+		device->uleds[i] = uled_reset;
+	}
+
+	return 0;
+}
+
+/***************************************************************************//**
+ * @brief Writes to the selected register on the device.
+ * @param device - The device structure.
+ * @param reg_addr - The register sub-address.
+ * @param reg_data - The data to write into the register (8-bits wide)
+ * @return 0 in case of success, negative error code otherwise.
+*******************************************************************************/
+int ltc3220_reg_write(struct ltc3220_dev *device, uint8_t reg_addr,
+		      uint8_t reg_data)
+{
+	if (reg_addr > LTC3220_REG_BLINK_GRAD)
+		return -EINVAL;
+
+	uint8_t data_buffer[2] = {0};
+
+	data_buffer[0] = reg_addr;
+	data_buffer[1] = reg_data;
+
+	return no_os_i2c_write(device->i2c_desc, data_buffer, 2, 1);
+}
+
+/***************************************************************************//**
+ * @brief Sets the selected ULED register's operating mode
+ * @param device - The device structure.
+ * @param uled_number - The ULED to select (1-indexed).
+ * @param mode - The mode to set the ULED to.
+ * @return 0 in case of success, negative error code otherwise.
+*******************************************************************************/
+int ltc3220_set_uled_mode(struct ltc3220_dev *device, uint8_t uled_number,
+			  enum ltc3220_uled_mode mode)
+{
+	int ret;
+	if (uled_number > LTC3220_REG_END_ULED)
+		return -EINVAL;
+
+
+	ret = ltc3220_update_reg_uled(device, uled_number, mode,
+				      device->uleds[uled_number - 1].current_level);
+	if (ret)
+		return ret;
+
+	device->uleds[uled_number - 1].mode = mode;
+
+	return 0;
+}
+
+/***************************************************************************//**
+ * @brief Sets the selected ULED register's current strength
+ * @param device - The device structure.
+ * @param uled_number - The ULED to select (1-indexed).
+ * @param current_level - The current strength from 0 to 20mA to set on the
+ * corresponding ULED pin (64 levels)
+*******************************************************************************/
+int ltc3220_set_uled_current(struct ltc3220_dev *device, uint8_t uled_number,
+			     uint8_t current_level)
+{
+	int ret;
```

Comment:
ditto. separate declaration from the rest of the code. check all occurences throughout the code.